shims/promise.js

```diff
@@ -1,4 +1,21 @@
-const RSVP = window.RSVP;
+/* globals require */
+
+function isNodeLikeEnvironment() {
+  const windowAbsent = typeof window === 'undefined';
+  const requirePresent = typeof require === 'function';
+
+  return windowAbsent && requirePresent;
+}
+
+let RSVP;
+
+if (isNodeLikeEnvironment()) {
+  const localRequire = require;
+
+  RSVP = localRequire('rsvp');
+} else {
+  RSVP = window.RSVP;
+}
```

Comment:
hmm... They are used in tests [here](https://github.com/Shopify/js-buy-sdk/blob/master/tests/unit/adapters/listings-adapter-test.js#L4) and [here](https://github.com/Shopify/js-buy-sdk/blob/master/tests/unit/api/shop-client-test.js#L5)